.circleci/config.yml

```diff
@@ -171,16 +272,28 @@ jobs:
           key: external-{{ arch }}-{{ .Branch }}-{{ epoch }}
           paths:
             - external
-
       - run: ccache --show-stats
+      - run:
+          name: rename artifacts
+          command: |
+            cp build/iroha-*.tar.gz build/iroha-darwin.tar.gz
+            # if IROHA_VERSION is not set, then CIRCLE_BUILD_NUM
+            echo ${IROHA_VERSION:-$CIRCLE_BUILD_NUM} > build/version.txt
+      - store_artifacts:
+          path: build/iroha-darwin.tar.gz
+          destination: iroha-darwin.tar.gz
+      - store_artifacts:
+          path: build/version.txt
+          destination: version.txt
+
      
 
 workflows:
   version: 2
   full_pipeline:
     jobs:
-      # macos-build/build -> sonar -> dockerize
-      - macos-build
+      # build-macos-release + build -> sonar(pr/release)
+      - build-macos-release
```

Comment:
It is for time-efficiency.
You should build iroha for linux both in debug (for coverage) and release (for dockerization). 
But, for macos you have to build only once (either debug/release is fine). Since we now collect build artifacts after every build, it is logical to build in release.

Now for every build:
- linux (debug) + coverage + analysis steps
- macos (release) + store artifacts (if you do not want to build branch by yourself, but want to use iroha binaries) steps

For release builds (push to develop, master, tag):
- linux (debug) + coverage + analysis
- linux (release) + dockerization + store artifacts
- macos (release) + store artifacts

Reply:
![image](https://user-images.githubusercontent.com/1867551/32697968-eab42722-c7ac-11e7-9aa3-c9810fcbb1bd.png)

It literally costs nothing.

Reply:
Can we measure time of build step with tar package and without it?

Reply:
Don't really sure that zipping would cost any significant time

Reply:
Yes, it costs like 10 seconds or so, for all `package` steps